keras/applications/convnext.py

```diff
@@ -0,0 +1,662 @@
+    self.drop_path = (
+      StochasticDepth(drop_path_rate, name=name + "_stochastic_depth")
+      if drop_path_rate > 0.0
+      else layers.Activation("linear", name=name + "_identity")
+    )
+
+  def call(self, inputs):
+    x = inputs
+
+    x = self.depthwise_conv_1(x)
+    x = self.layer_norm(x)
+    x = self.pointwise_conv_1(x)
+    x = self.act_fn(x)
+    x = self.pointwise_conv_2(x)
+
+    if self.gamma is not None:
+      x = self.gamma * x
+
+    return inputs + self.drop_path(x)
+
+  def get_config(self):
+    config = {
+      "projection_dim": self.projection_dim,
+      "drop_path_rate": self.drop_path_rate,
+      "layer_scale_init_value": self.layer_scale_init_value,
+    }
+    return config
+
+
+def PreStem(name=None):
+  """Normalizes inputs with ImageNet-1k mean and std.
+
+  Args:
+    name (str): Name prefix.
+
+  Returns:
+    Normalized tensor.
+  """
+  if name is None:
+    name = "prestem" + str(backend.get_uid("prestem"))
+
+  def apply(x):
+    x = layers.Normalization(
+      mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+      variance=[(0.229 * 255) ** 2, (0.224 * 255) ** 2, (0.225 * 255) ** 2],
+      name=name + "_prestem_normalization"
+    )(x)
+    return x
+
+  return apply
+
+
+def Head(num_classes=1000, name=None):
+  """Implementation of classification head of RegNet.
+
+  Args:
+    num_classes: number of classes for Dense layer
+    name: name prefix
+
+  Returns:
+    Output logits tensor.
+  """
+  if name is None:
+    name = str(backend.get_uid("head"))
+
+  def apply(x):
+    x = layers.GlobalAveragePooling2D(name=name + "_head_gap")(x)
+    x = layers.LayerNormalization(
+      epsilon=1e-6, name=name + "_head_layernorm")(x)
+    x = layers.Dense(num_classes, name=name + "_head_dense")(x)
+    return x
+
+  return apply
+
+
+def ConvNeXt(depths,
+  projection_dims,
+  drop_path_rate=0.0,
+  layer_scale_init_value=1e-6,
+  default_size=224,
+  model_name="convnext",
+  include_preprocessing=True,
+  include_top=True,
+  weights=None,
+  input_tensor=None,
+  input_shape=None,
+  pooling=None,
+  classes=1000,
+  classifier_activation="softmax"):
+  """Instantiates ConvNeXt architecture given specific configuration.
+
+  Args:
+    depths: An iterable containing depths for each individual stages.
+    projection_dims: An iterable containing output number of channels of
+    each individual stages.
+    drop_path_rate: Stochastic depth probability. If 0.0, then stochastic depth
+      won't be used.
+    layer_scale_init_value: Layer scale coefficient. If 0.0, layer scaling won't
+      be used.
+    default_size: Default input image size.
+    model_name: An optional name for the model.
+    include_preprocessing: boolean denoting whther to include preprocessing in
+      the model. When `weights="imagenet"` this should be always set to True.
+      But for other models (e.g., randomly initialized) users should set it
+      to False and apply preprocessing to data accordingly.
```

Comment:
@LukeWood added an extended note regarding preprocessing so that our users are aware.